### DISCUSSION TITLE
How do I use html-react-parser with <Highlight attribute="description" hit={hit}  /> ?

### DISCUSSION THREAD

**Discussion by da-kicks-87** [algolia/instantsearch#5749] [Q&A]
Hello ,

I am using `react-instantsearch-hooks-web`

In my DB i have HTML string as the description. I need to parse it so it output as HTML and not output the tags.

I want to use the component `<Highlight attribute="description" hit={hit} />`

The attribute description needs to be parsed. In the rest of my app i am using the libray `html-react-parser`

I have tried this, but no luck.
    

`const modifiedDescription = parse(hit.description);`

`<Highlight attribute="description" hit={{ ...hit, description: modifiedDescription }}  />`

It still output the HTML tags in the browser.

Does anyone know how to solve this?

**Comment by Haroenv** [comment]
The highlight component specifically escapes html, as they are generally a danger. You can either:

- use dangerouslySetInnerHTML instead of Highlight `<div dangerouslySetInnerHTML={{__html: hit._highlightResult.description.value }} />`
- use html react parser: `{parse(hit._highlightResult.description.value)}`

Both of those options will require you to pass `escapeHTML: false` on any `Hit` components you have, otherwise html still be escaped.

**Comment by da-kicks-87** [comment]
Sorry that still doesn't work.

I have tried this:

`{parse(hit._highlightResult.description.value)}
<Highlight attribute="description" escapeHTML={false} hit={hit}  />`

Any ideas? 
Can you point to me where is the documentation where I can find more information about how to access the values passed to these components?

**Reply by Haroenv** [reply]
it's Hit which should get escapeHTML=false, not Highlight: https://www.algolia.com/doc/api-reference/widgets/hits/react-hooks/#widget-param-escapehtml

**Comment by da-kicks-87** [comment]
Cool, I am getting closer. I am able to output the parsed result with this:

`  <InfiniteHits hitComponent={Hit} escapeHTML={false}/>`
`{parse(hit._highlightResult.description.value)} `

However  want the result pass into the `<Hightlights>` component.
Would I have to use the `transformItems` prop  and use it on `<InfiniteHits>`?

I noticed the once adding `escapeHTML={false}`to `<InfiniteHit>` to attribute name gets distorted output in some cases.

See screen shot below. The bold title is what is being distorted. It should say 202 Pendant.
I looks like the names that start with numbers the distortion occurs.

![Screen Shot 2023-07-12 at 1 36 29 PM](https://github.com/algolia/instantsearch/assets/70115804/60ee30e0-582f-4de6-89c3-9d528f6e744b)

**Reply by Haroenv** [reply]
If you want to use plain html you can do either:

- replace `__ais-highlight__` with `<mark>` and `__/ais-highlight__` with `</mark>` using a regex
- set `highlightPreTag` and `highlightPostTag` to `<mark>` and `</mark>` using `<Configure highlightPreTag="<mark>" highlightPostTag="</mark>"/>`

**Comment by da-kicks-87** [comment]
Using `<Configure highlightPreTag="<mark>" highlightPostTag="</mark>"/>` solved the distorting onf the `name` title. 

What about cleaning up the code to pass the parsed result to the <Hightlights> component?
I am currently using this to output the parsed description: ` {parse(hit._highlightResult.description.value)}   `

Thank you!

**Reply by Haroenv** [reply]
What do you mean cleaning up? isn't `parse` sufficiently clean? or is it hit._highlightResult.description.value that you don't want to see?

**Comment by da-kicks-87** [comment]
The output to the browser is how i want it. It's the code that I have questions about.
I am using `hit._highlightResult.description.value` instead of `<Highlight attribute="description" hit={hit}  />`  .

My question is shouldn't I be passing the parsed value to `<Highlight attribute="description" hit={hit}  />`  . ?

**Reply by Haroenv** [reply]
No, highlight is specifically made to avoid innerHTML or custom html, so it will render everything that's not the "highlight tags" as strings, which isn't what you want in your case, even if it's processed with `html-parser-react`

**Comment by da-kicks-87** [comment]
Ok. I think I understand. 

Isn't it common the have descriptions with html? I feel doing this process should have been simpler.
It would be good to make mention of this situation in the documentation.

**Reply by Haroenv** [reply]
No, it's not common to have html in the items as it's a possible attack vector if anything is user-generated. We recommend using a more structured format instead

**Comment by da-kicks-87** [comment]
The product description comes from a CMS Rich Text Editor field (CKEditor 5). We wanted user to be able to format the text and add link etc... That why it outputs HTML.
